logging/src/commonMain/kotlin/SensitiveAwareLogger.kt

```diff
@@ -0,0 +1,82 @@
+/*
+ Copyright 2025 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.logging
+
+import com.splendo.kaluga.logging.SensitiveAwareLogger.SensitiveAwareContext
+
+/** Logger that correctly handles a sensitive data. */
+interface SensitiveAwareLogger {
```

Comment:
decided not to go by extending a `Logger` as I'd like to try a slightly different API:
* `SensitiveAwareLogger` requires a tag and knows with which tag it's created
* it also requires a message
* allows creating "child" loggers, that use the same settings

this is done to prevent dragging 2 loggers and tags across classes